libpod/container_internal.go

```diff
@@ -635,6 +636,22 @@ func (c *Container) init(ctx context.Context) error {
 		return err
 	}
 
+	// To support daemon-less container engines, we need to defer loading
+	// of the default AppArmor profile until this point.  All containers,
+	// new and old ones, must be initialzed when being (re)started.  Old
+	// containers, however, might still reference and old version of the
+	// profile or the new one might not yet be loaded.
+	//
+	// In case a newer version of the default profile is available, update
+	// it the spec.
+	if apparmor.IsEnabled() {
```

Comment:
This would probably be better in `generateSpec()` (we can directly add the Apparmor profile to the generator there instead of setting it in c.config.Spec)

Reply:
Also - does this need to be disabled for rootless containers?

Reply:
Let me get the checks back to be on the safe side.

Reply:
done